### PR TITLE
Improvements to Form Detail template

### PR DESCRIPTION
**Pull request by ncksllvn**
## Description
Attempts the changes as outlined in https://github.com/department-of-veterans-affairs/va.gov-team/issues/10606#issuecomment-657660113

- Form PDF links should download to the tray
    - Created a polyfill for IE to add support for `download` attribute
- Add business logic for the `related to` field
- Update headings for the related forms section

## Testing done
Ran preview server locally to test against Drupal data

## Screenshots

This main visual difference is the `Related to` piece using different labels for the benefit hubs. 

image

## Acceptance criteria
- [ ] Updates are made per comment

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/site/layouts/va_form.drupal.liquid [hunk 1]
```diff
@@ -24,38 +24,46 @@
...
-        {% if fieldBenefitCategories.length > 0 or fieldVaFormRevisionDate %}
-          <dl>
-            {% if fieldBenefitCategories.length > 0 %}
-              <div>
-                <dt class="vads-u-font-weight--bold vads-u-display--inline">
-                  Related to:
-                </dd>
-                <dd class="vads-u-display--inline">
+        <dl>
+          <div class="vads-u-margin-y--1">
```
Comment (ncksllvn) [inline src/site/layouts/va_form.drupal.liquid:28]: > line break between Related to and Form last updated - needs para break or more line space

File: src/site/layouts/va_form.drupal.liquid [hunk 2]
```diff
@@ -24,38 +24,46 @@
...
+          {% endif %}
+        </dl>
 
         {% if fieldVaFormUsage %}
           <h3>When to use this form</h3>
           {{ fieldVaFormUsage.processed }}
         {% endif %}
 
-        <a href="{{ fieldVaFormUrl.uri }}">Download VA Form {{ fieldVaFormNumber }} (PDF)</a>
+        <a href="{{ fieldVaFormUrl.uri }}" download>Download VA Form {{ fieldVaFormNumber }} (PDF)</a>
```
Comment (ncksllvn) [inline src/site/layouts/va_form.drupal.liquid:66]: > All Download links on the form detail page - make them initiate download, same as with download links on forms search

We're gonna run into the same problem as on Find Forms though where IE doesn't support the `download` attribute. We'll have to circle back around.

File: src/site/layouts/va_form.drupal.liquid [hunk 3]
```diff
@@ -72,12 +80,15 @@
...
-                  <h4>
-                    VA Form {{ vaForm.entity.fieldVaFormNumber }}
-                    <span class="vads-u-display--block">{{ vaForm.entity.fieldVaFormName }}</span>
-                  </h4>
+                  <hgroup>
+                    <h4>VA Form {{ vaForm.entity.fieldVaFormNumber }}</h4>
+                    <h5 class="vads-u-font-size--base vads-u-margin-top--0">
+                      <span class="vads-u-visibility--screen-reader">Form name:</span> {{ vaForm.entity.fieldVaFormName }}
+                    </h5>
+                  </hgroup>
```
Comment (ncksllvn) [inline src/site/layouts/va_form.drupal.liquid:88]: > Under the Related forms and instructions section: Can you add vertical/para break space after the form number and the form name? (Or (prefer) take the form name out of the H4 and make it strong text. If that's not 508-compliant, then 2nd preference, make the form number and form title each individual H4s?

File: src/site/layouts/va_form.drupal.liquid [hunk 4]
```diff
@@ -24,38 +24,46 @@
...
-                </dd>
-              </div>
-            {% endif %}
-          </dl>
-        {% endif %}
+                {% endif %}
+
+              {% else %}
+                {{ fieldVaFormAdministration.entity.entityLabel }}
+              {% endif %}
```
Comment (ncksllvn) [inline src/site/layouts/va_form.drupal.liquid:45]: > Related to field values - See content doc here for FE copy mapping - page 2 shows the copy that should be used to display for each data value.

There's a lot of logic here. Still a WIP.

File: src/applications/find-forms/components/SearchResult.jsx [hunk 5]
```diff
@@ -1,39 +1,19 @@
...
   const isSameOrigin = form?.attributes?.url.startsWith(window.location.origin);
   const isPDF = form?.attributes?.url.toLowerCase().includes('.pdf');
-  const isInternetExplorer = !!navigator.msSaveBlob;
 
   if (!isSameOrigin || !isPDF) {
     // Just open in a new tab if we'd otherwise hit a CORS issue or if the form URL isn't a PDF.
     linkProps.target = '_blank';
-  } else if (isInternetExplorer) {
-    // IE doesn't support the HTML download attribute.
-    linkProps.onClick = onDownloadClick(form.attributes.url);
```
Comment (ncksllvn) [inline src/applications/find-forms/components/SearchResult.jsx:36]: This was all chunked out into a polyfill for the `download` attribute in IE

File: src/site/layouts/va_form.drupal.liquid (hunk 2 above)
Comment (ncksllvn) [reply src/site/layouts/va_form.drupal.liquid:68]: Fixed by chunking out the Find Forms download logic into a polyfill

**Review by kelsonic — APPROVED** [review]
Looks really nice! 🎉 ✨

**Review by f1337 — APPROVED** [review]
Looks good to me! :thumbsup:
Sean The Sheep giving thumbs up and smiling

**Review by U-DON — APPROVED** [review]

**Pull request merged**